### PR TITLE
fix: Send attachments as multiple messages for whatsapp channel

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

Fixes #https://github.com/chatwoot/product/issues/704

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -701,18 +701,30 @@ export default {
         if (this.isSignatureEnabledForInbox && this.messageSignature) {
           newMessage += '\n\n' + this.messageSignature;
         }
-        const messagePayload = this.getMessagePayload(newMessage);
+
+        if (this.isWhatsAppChannel) {
+          this.sendMessageAsMultipleMessages(newMessage);
+        } else {
+          const messagePayload = this.getMessagePayload(newMessage);
+          this.sendMessage(messagePayload);
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:709]: This changes the sequence of things. Earlier `clearMessage` was called after `sendMessage`, now it's before. Is that fine?

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 2]
```diff
@@ -701,18 +701,30 @@ export default {
...
+        }
 
         this.clearMessage();
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:712]: Line 717 has another `clearMessage` call, we can probably get rid of one

**Review by scmmishra — CHANGES_REQUESTED** [review]
The component is already over 1000 lines, which could be better. Perhaps we could move the Whatsapp-related utilities to a separate js file. It will make the footprint of this shorter and we will be able to test this easily

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue (hunk 2 above)
Comment (nithindavid) [reply app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:712]: I have removed the extra clearMessage call.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue (hunk 1 above)
Comment (nithindavid) [reply app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:709]: This was done as we are iterating over each attached message and sending it. Calling clearMessage would clear the `attachedFiles` state variable. Let me know your thoughts

**Review by iamsivin — APPROVED** [review]
Code LGTM

**Review by scmmishra — APPROVED** [review]
> The component is already over 1000 lines, which could be better. Perhaps we could move the Whatsapp-related utilities to a separate js file. It will make the footprint of this shorter and we will be able to test this easily

About this, let's pick this up as a separate task, ReplyBox can have more tests. Approving this PR for now

**Comment by pranavrajs** [comment]
@nithindavid @scmmishra This PR introduces a major bug. All the private notes added to the WhatsApp channel are sent to the customer.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the comment above.

**Pull request merged**